### ISSUE TITLE
Invalid snapshot when model prop changed in onAttachedToRootStore

### ISSUE THREAD

**Issue by sisp** [xaviergonz/mobx-keystone#74]
I think a test case that reproduces this problem explains it best:

```ts
import {
  getSnapshot,
  Model,
  model,
  modelAction,
  prop,
  registerRootStore,
} from 'mobx-keystone';

test('snapshot', () => {
  @model('#74/A')
  class A extends Model({
    value: prop<number>(),
  }) {
    public onAttachedToRootStore() {
      this.value = 1; // but, e.g, this works: setTimeout(() => { this.value = 1; })
    }
  }

  @model('#74/Store')
  class Store extends Model({
    all: prop<A[]>(() => []),
  }) {
    @modelAction
    public add(a: A): void {
      this.all.push(a);
    }
  }

  const store = registerRootStore(new Store({}));
  store.add(new A({ value: 0 }));

  expect(getSnapshot(store).all).toHaveLength(store.all.length);
});
```

The state is correct, but the snapshot is wrong. Unfortunately, I haven't been able to determine the cause of this error.

**Comment by xaviergonz** [comment]
Thanks! should be fixed in 0.28.2, please reopen if not.
The fix adds a slight change in behaviour though. onAttachedToRootStore and its disposer will now run after the current action / runUnprotected block is finsihed (it will still run synchronously though).

**Comment by sisp** [comment]
Works perfectly, thanks a lot!
